app/src/main/cpp/skyline/vfs/ticket.cpp

```diff
@@ -0,0 +1,39 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2021 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include "ticket.h"
+
+namespace skyline::vfs {
+    enum class SignatureType : u32 {
+        Rsa4096Sha1 = 0x010000,
+        Rsa2048Sha1 = 0x010001,
+        EcdsaSha1 = 0x010002,
+        Rsa4096Sha256 = 0x010003,
+        Rsa2048Sha256 = 0x010004,
+        EcdsaSha256 = 0x010005
+    };
+
+    Ticket::Ticket(const std::shared_ptr<vfs::Backing> &backing) {
+        auto type{backing->Read<SignatureType>()};
+
+        size_t offset;
+        switch (type) {
+            case SignatureType::Rsa4096Sha1:
+            case SignatureType::Rsa4096Sha256:
+                offset = 0x240;
+                break;
+            case SignatureType::Rsa2048Sha1:
+            case SignatureType::Rsa2048Sha256:
+                offset = 0x140;
+                break;
+            case SignatureType::EcdsaSha1:
+            case SignatureType::EcdsaSha256:
+                offset = 0x80;
+                break;
+            default:
+                throw exception("Could not find valid signature type 0x{0:X}", type);
```

Comment:
Is there any reason to use `0x{0:X}` rather than `0x{:X}`? `libfmt` will assume the indices are in an incrementing order when you don't specify one, which you should only do when you print a single argument twice.

Reply:
Oh sure, I didn't really know what to put here, and I forgot to ask you on discord, will change to the one you suggested.